src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -300,6 +292,35 @@ public async Task InitializeAsync(CancellationToken cancellationToken = default)
             }
         }
 
+        /// <summary>
+        /// Appends specific functions worker runtime version after runtime stack
+        /// </summary>
+        /// <returns>A string contains specific runtime stack (e.g. python-3.6, dotnet-~2) or single stack</returns>
+        private string GetPreciseRuntimeStack(string runtime)
+        {
+            string runtimeStack = runtime?.ToLower() ?? string.Empty;
+            string preciseRuntime = string.Empty;
+            switch (runtimeStack)
+            {
+                case "dotnet":
+                    // Get Dotnet version from FUNCTIONS_EXTENSION_VERSION
+                    preciseRuntime = $"dotnet-{_environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsExtensionVersion)}";
+                    break;
```

Comment:
This won't work for Linux.

Reply:
Yea, the current behavior will return `dotnet` instead of `dotnet-~2` if this the FUNCTIONS_EXTENSION_VERSION is not set in Linux Consumption. Please notify me for the change necessary.